tests/lib/rules/require-expect.js

```diff
@@ -125,12 +130,24 @@ ruleTester.run("require-expect", rule, {
             options: ["except-simple"]
         },
 
+        // When the test body itself is using an arrow function, make sure we still correctly-detect top-level expect.
```

Comment:
This is the only test case that was broken before since this bug only affected the `except-simple` config. I added corresponding test cases to the other configs regardless.